### PR TITLE
Add a CellCollection mutable sequence wrapping a CellArray

### PR DESCRIPTION
**Pull request by itziakos**

### CODE REVIEW

**Comment by itziakos** [comment]
@dpinte this should be easier to review

File: simphony_mayavi/core/cell_collection.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            The tvtk object to wrap. Default value is an empty
+            tvtk.CellArry.
+
+        """
+        if cell_array is None:
+            cell_array = tvtk.CellArray()
+        self._cell_array = cell_array
+
+    def __len__(self):
+        """ The length of the
```
Comment (sjagoe) [inline simphony_mayavi/core/cell_collection.py]: Truncated docstring

File: simphony_mayavi/core/cell_collection.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+import numpy
+from tvtk.api import tvtk
+
+
+#### FIXME This is a good candidate class for mayavi.
+
+
+class CellCollection(MutableSequence):
+    """ A mutable sequence of cells rapping a tvtk.CellArray.
```
Comment (sjagoe) [inline simphony_mayavi/core/cell_collection.py]: `wrapping`

File: simphony_mayavi/core/cell_collection.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        points = tvtk.IdList()
+        self._cell_array.get_cell(location, points)
+        return tuple(points)
+
+    def __setitem__(self, index, value):
+        """ Update the connectivity list for cell at ``index``.
+
+        .. note::
+
+           If the size of the connectivity list changes as slower
```
Comment (sjagoe) [inline simphony_mayavi/core/cell_collection.py]: `a slower path`

File: simphony_mayavi/core/cell_collection.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        .. note::
+
+           This operation will need to create temporary arrays in order
+           to keep the data info consistent.
+
+
+        """
+        length = len(self)
+        cells = self._cell_array
+        location = self._cell_location(index)
```
Comment (sjagoe) [inline simphony_mayavi/core/cell_collection.py:84]: This should move after the bounds check.

File: simphony_mayavi/core/cell_collection.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+            array = numpy.r_[left, new_cell, right]
+            cells.set_cells(length + 1, array)
+        elif index >= length:
+            cells.insert_next_cell(value)
+        else:
+            raise IndexError('{} is out of index range'.format(index))
+
+    # Private methods ######################################################
+
+    def _add_arrays(self, arrays):
```
Comment (sjagoe) [inline simphony_mayavi/core/cell_collection.py]: Unused.

File: simphony_mayavi/core/cell_collection.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+           to keep the data info consistent.
+
+
+        """
+        length = len(self)
+        cells = self._cell_array
+        location = self._cell_location(index)
+        npoints = cells.data[location]
+        start = location + 1
+        if 0 <= index < length:
```
Comment (sjagoe) [inline simphony_mayavi/core/cell_collection.py]: Move this check to the first location in `_cell_location()`

File: simphony_mayavi/core/tests/test_cell_collection.py [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        # when/then
+        with self.assertRaises(IndexError):
+            collection[5]
+
+    def test_getitem_with_long_int_index(self):
+        # given
+        cells = self.cells
+        collection = CellCollection(self.vtk)
+
+        # when/then
```
Comment (sjagoe) [inline simphony_mayavi/core/tests/test_cell_collection.py:55]: I don't understand many of the `when/then` comments.

Comment (itziakos) [inline simphony_mayavi/core/tests/test_cell_collection.py:55]: This one is correct, I have updated the other places were when/then was used by mistake

**Comment by itziakos** [comment]
Thanks @sjagoe

**Pull request merged**